numpy_ml/linear_models/naive_bayes.py

```diff
@@ -1,128 +1,211 @@
-            Input data
-
-        mean: ndarray of shape (n_classes,)
-            mean of each variance
-        
-        sigma: ndarray of shape (n_classes,)
-            variance of each class
-        
-        prior :  ndarray of shape (n_classes,)
-                probability of each class
+        Parameters
+        ----------
+        X: :py:class:`ndarray <numpy.ndarray>` of shape `(N, M)`
+            A dataset of `N` examples, each of dimension `M`
 
-        returns
+        Returns
         -------
-        joint_log_likelihood : ndarry of shape (n_samples,)
-            joint log likelihood of data
-        
+        log_posterior : :py:class:`ndarray <numpy.ndarray>` of shape `(N, K)`
+            Unnormalized log posterior probability of each class for each
+            example in `X`
         """
+        K = len(self.labels)
+        log_posterior = np.zeros((X.shape[0], K))
+        for i in range(K):
+            log_posterior[:, i] = self._log_class_posterior(X, i)
+        return log_posterior
 
-        prob = -self.n_features / 2 * np.log(2 * np.pi) - 0.5 * np.sum(
-            np.log(sigma )
-            )
-        prob -= 0.5 * np.sum(np.power(X -mean, 2) / (sigma), 1)
-
-        joint_log_likelihood = prior + prob
-        return joint_log_likelihood
+    def _log_class_posterior(self, X, class_idx):
+        r"""
+        Compute the (unnormalized) log posterior for the label at index
+        `class_idx` in :py:attr:`labels`.
 
+        Notes
+        -----
+        Unnormalized log posterior for example :math:`\mathbf{x}_i` and class
+        :math:`c` is::
 
+        .. math::
 
+            \log P(y_i = c \mid \mathbf{x}_i, \theta)
+                &\propto \log P(y=c \mid \theta) +
+                    \log P(\mathbf{x}_i \mid y_i = c, \theta) \\
+                &\propto \log P(y=c \mid \theta)
+                    \sum{j=1}^M \log P(x_j \mid y_i = c, \theta)
 
+        In the Gaussian naive Bayes model, the feature likelihood for class
+        :math:`c`, :math:`P(\mathbf{x}_i \mid y_i = c, \theta)` is assumed to
+        be normally distributed
 
+        .. math::
 
+            \mathbf{x}_i \mid y_i = c, \theta \sim \mathcal{N}(\mu_c, \Sigma_c)
 
 
+        Parameters
+        ----------
+        X: :py:class:`ndarray <numpy.ndarray>` of shape `(N, M)`
+            A dataset of `N` examples, each of dimension `M`
+        class_idx : int
+            The index of the current class in :py:attr:`labels`
 
-    
+        Returns
+        -------
+        log_class_posterior : :py:class:`ndarray <numpy.ndarray>` of shape `(N,)`
+            Unnormalized log probability of the label at index `class_idx`
+            in :py:attr:`labels` for each example in `X`
+        """
+        P = self.parameters
+        mu = P["mean"][class_idx]
+        prior = P["prior"][class_idx]
+        sigsq = P["sigma"][class_idx]
+
+        # log likelihood = log X | N(mu, sigsq)
```

Comment:
@sfsf9797 this is what I believe the proper log likelihood + log class posterior calc should be. Let me know if you agree!